### PR TITLE
[UI] Improved my-profile UI and functionality

### LINKED ISSUE #1467: [UI idea] Update my profile page
**Idea incl level**
Currently the my profile page contains the user information as well as the logout button and the option to change the user password. I would like to expand this page to contain more useful information. Will have to be brainstormed about, but for example:

- Total programs saved
- Maximum level saved
- Current classes (Students have no idea in which classes they are enrolled)

Also I would like to improve the overall UI to better align with the rest of the Hedy website. Especially shifting the logout button to a different place. In aligned with #1466 it would be great to implement a visual nice way to show the logout button when a user is logged in. As icons are already implemented, we might even be able to do this with a nice button.

### LINKED ISSUE #1478: [UI idea] When logged in, set language automatically to user preferred language
**Idea incl level**
Currently the UI is set to a user selected language. When logging in the language will stay the same. However, when we built upon #1477 the language selection will disappear from the menubar when logged in. In this case we would like the user to selected a preferred language in the profile settings to use every time the user is logged in. This should be aligned with #1467 to make sure users are able to change their preferred language on the profile page.

### PR DESCRIPTION
**Pull request by TiBiBa**
**Description**
In this PR we clean-up the my-profile page and improve functionality. Currently the my-profile page only has functionality to change personal settings, password and/or delete the account. In this PR we introduce user statistics, class overview and a section-structure creating a more clean page. Shortly the followed functionality is introduced:

- Collapsable tabs, containing sections of relevant information
- User statistics (currently only amount of programs saved, can be expanded in future work)
- Self-removal of classes (student where till this day unable to leave a class)
- Setting preferred language (removing language selector from menubar when logged in)

**Old UI**
Schermafbeelding 2021-12-08 om 17 29 41

**New UI**
Schermafbeelding 2021-12-08 om 18 08 42

**Fix for**
This PR fixed #1467 and fixed #1478.

**How to test**

Describe how to run your code to see it is action. See this https://github.com/Felienne/hedy/pull/880#issue-1016304308 for an example.

**Checklist**

If you're unsure about any of these, don't hesitate to ask. We're here to help!
- [ ] Links to an existing issue or discussion (if not, create an issue first)
- [ ] Describes changes clear in the format above (present tense, no subject)
- [ ] Has a "how to test" section
- [ ] Only one thing is done in this pull request (specifically please try to refrain from mixing textual changes to the yamls from code changes)

### CODE REVIEW

**Comment by TiBiBa** [comment]
Added (temporary) header message for current accounts that do not have a preferred language set yet. When a user is either nog logged in or has a language set the message will not be shown.

Schermafbeelding 2021-12-08 om 20 13 42

**Review by Felienne — CHANGES_REQUESTED** [review]
Hi @TiBiBa!

As discussed we want a few e2e tests for these new features and an update to the teacher's manual.

**Comment by TiBiBa** [comment]
> Hi @TiBiBa!
> 
> As discussed we want a few e2e tests for these new features and an update to the teacher's manual.

Hope to get it done today! Otherwise it will be Sunday or Monday (won't work tomorrow).

**Review by Felienne — CHANGES_REQUESTED** [review]
Thanks for the additional test @TiBiBa! One of them is failing still unfortunately :)

**Comment by Felienne** [comment]
> Added (temporary) header message for current accounts that do not have a preferred language set yet. When a user is either nog logged in or has a language set the message will not be shown.
> 
> Schermafbeelding 2021-12-08 om 20 13 42

This is a bit of an "angry" looking message... Can we make it a bit more friendly? 

And will all kids get that, or just the ones where we were not able to guess a language based on their programs (not sure if we ended up implementing a "guesser")

File: coursedata/texts/nl.yaml [hunk 1]
```diff
@@ -120,6 +120,8 @@ Auth:
     password_repeat: "Herhaal je wachtwoord"
     birth_year: "Geboortejaar (optioneel)"
     gender: "Geslacht (optioneel)"
+    preferred_language: "Voorkeurstaal"
+    set_preferred_lang: "Hedy accounts kunnen vanaf nu een voorkeurstaal keizen. Jij hebt dit nog niet gedaan: Stel dit in voor jouw account bij \"Mijn profiel\""
```
Comment (Felienne) [inline coursedata/texts/nl.yaml:124]:
```suggestion
    set_preferred_lang: "Hedy accounts kunnen vanaf nu een voorkeurstaal kiezen. Jij hebt dit nog niet gedaan: Stel dit in voor jouw account bij \"Mijn profiel\""
```

File: static/js/app.ts [hunk 2]
```diff
@@ -1027,6 +1027,17 @@ export function toggle_developers_mode(example_programs: boolean) {
   }
 }
 
+export function load_profile(username: string, mail: string, birth_year: number, gender: string, country: string) {
+  $('#profile').toggle();
+  if ($('#profile').is(":visible")) {
```
Comment (Felienne) [inline static/js/app.ts:1032]: Zou de gekozen taal dan hier niet ook bij moeten staan?

**Comment by TiBiBa** [comment]
> > Added (temporary) header message for current accounts that do not have a preferred language set yet. When a user is either nog logged in or has a language set the message will not be shown.
> > Schermafbeelding 2021-12-08 om 20 13 42
> 
> This is a bit of an "angry" looking message... Can we make it a bit more friendly?
> 
> And will all kids get that, or just the ones where we were not able to guess a language based on their programs (not sure if we ended up implementing a "guesser")

This message is only shown to logged in user without a preferred language set in their profile settings. If they configure one it will disappear.

File: static/js/app.ts (hunk 2 above)
Comment (TiBiBa) [reply static/js/app.ts:1032]: Als ik het me goed herinner halen we die uit de session en zetten hem dan in de form

Comment (TiBiBa) [reply static/js/app.ts:1032]: Deze functie wordt alleen gebruikt voor het handig parsen van de session values naar het html form

**Comment by Felienne** [comment]
> > > Added (temporary) header message for current accounts that do not have a preferred language set yet. When a user is either nog logged in or has a language set the message will not be shown.
> > > Schermafbeelding 2021-12-08 om 20 13 42
> > 
> > 
> > This is a bit of an "angry" looking message... Can we make it a bit more friendly?
> > And will all kids get that, or just the ones where we were not able to guess a language based on their programs (not sure if we ended up implementing a "guesser")
> 
> This message is only shown to logged in user without a preferred language set in their profile settings. If they configure one it will disappear.

Yeah I git it is only shown temporarily but it still might make a user feel like they did something wrong. Maybe we can make it yellow instead of red? And phrased a bit more like: "We now support default languages" instead "you have not done..." (hope this makes sense!

**Comment by TiBiBa** [comment]
Changed the UI on the warning message! However, Github does not seem to start checking? (Had this issue with the last 5 commits or so)

**Review by Felienne — APPROVED** [review]

File: static/js/auth.ts [hunk 3]
```diff
@@ -278,34 +285,6 @@ $ ('.auth input').get ().map (function (el) {
 });
 
 // We use GET /profile to see if we're logged in since we use HTTP only cookies and cannot check from javascript.
-$.ajax ({type: 'GET', url: '/profile'}).done (function (response) {
-  if (['/signup', '/login'].indexOf (window.location.pathname) !== -1) auth.redirect ('my-profile');
-
-  auth.profile = response;
```
Comment (rix0rrr) [inline static/js/auth.ts:284]: This were was a load-bearing statement.

I agree it's not fantastic, but fetching `/profile` using AJAX and then copying the response onto `auth.profile` is how the Save button knows you are logged in.

So either you need to put this back, or come up with a new way of setting the same property (or checking for the logged-in condition differently).

**Pull request merged**